### PR TITLE
[BUG] Fix On The Fly Issues with mesh layer

### LINKED ISSUE #28827: Mesh layer does not work properly with OTF reprojection
Author Name: **Saber Razmjooei** (@saberraz)
Original Redmine Issue: [21008](https://issues.qgis.org/issues/21008)
Affected QGIS version: 3.5(master)
Redmine category:data_provider/mdal
Assignee: Peter Petrik

---
I have the following datasets:
- Mesh layer (global coverage)
- Countries (shapefile)

It works well with the canvas projection set to the mesh projection. But with canvas set to 3857, the mesh shrinks to a small area and the quantity disappears.

---

- [3857.png](https://issues.qgis.org/attachments/download/14105/3857.png) (Saber Razmjooei)
- [4326.png](https://issues.qgis.org/attachments/download/14106/4326.png) (Saber Razmjooei)
- [21008.zip](https://issues.qgis.org/attachments/download/14107/21008.zip) (Saber Razmjooei)

### LINKED ISSUE #29330: MDAL: crashes when zooming on EPSG:4326
Author Name: **Paolo Cavallini** (@pcav)
Original Redmine Issue: [21513](https://issues.qgis.org/issues/21513)
Affected QGIS version: 3.6.0
Redmine category:mesh_layers
Assignee: Peter Petrik

---
* open QGIS 3.6
* type 'woorld' in the ccod box, to load a world map in 4326
* load the file: https://www.faunalia.eu/~paolo/20190307_190528_GFS_P25_.grb2
* QGIS crashes
All from Debian sid, up  to date.

---

- [Schermata del 2019-03-08 14-55-28.png](https://issues.qgis.org/attachments/download/14557/Schermata%20del%202019-03-08%2014-55-28.png) (Alberto Grava)

### PR DESCRIPTION
**Pull request by vcloarec**
fix #29330 and fix #28827

### CODE REVIEW

File: src/core/mesh/qgstriangularmesh.cpp [hunk 1]
```diff
@@ -120,6 +130,7 @@ bool QgsTriangularMesh::update( QgsMesh *nativeMesh, const QgsCoordinateTransfor
   mNativeMeshFaceCentroids.clear();
 
   // TRANSFORM VERTICES
+  QSet<int> invalidVertexIndexes;
```
Comment (PeterPetrik) [inline src/core/mesh/qgstriangularmesh.cpp]: unused? remove?

Comment (vcloarec) [reply src/core/mesh/qgstriangularmesh.cpp]: unused, so removed...

**Comment by nyalldawson** [comment]
Unrelated test failure -- don't get hung up on that one :wink:

**Comment by vcloarec** [comment]
@nyalldawson , azure is so long...

**Comment by nyalldawson** [comment]
@vcloarec it's a known issue... we can't enable the cache because of a bug/limitation in azure

**Pull request merged**